### PR TITLE
Fix sniffer

### PR DESCRIPTION
**Pull request by bandi13**
# Description
Using Linux, there is a 'linux cooked capture' layer that is added on, and the packets seem to contain two extra bytes.

Also reworked the argument handling to be able to individually add in arguments without needing to pass in a PCAP file. Now we can sniff live packets and set filters. Additionally, we can now define the output of the trace file.

# Testing
Able to read packet capture when using 'any' interface

### CODE REVIEW

**Comment by JacobBarthelmeh** [comment]
Assigning back to @bandi13. Failing one of the sniffer tests `snifftest static RSA failed`

File: sslSniffer/sslSnifferTest/snifftest.c [hunk 1]
```diff
@@ -1290,22 +1290,30 @@ int main(int argc, char** argv)
...
+            if (header->caplen > 40)  { /* min ip(20) + min tcp(20) */
                 packet        += frame;
-                header.caplen -= frame;
+                header->caplen -= frame;
             }
             else {
                 /* packet doesn't contain minimum ip/tcp header */
                 continue;
             }
+            int ptype = pcap_datalink(pcap);
```
Comment (dgarske) [inline sslSniffer/sslSnifferTest/snifftest.c:1308]: Avoid mixed-declaration. Move `int ptype` up. Also please add comment documenting what Linux SLL is and what the 2 bytes are.

**Review by dgarske — CHANGES_REQUESTED** [review]
Ask discussed yesterday. Works now with "any" interface, but not with "lo".

**Comment by bandi13** [comment]
retest this please

File: src/sniffer.c [hunk 2]
```diff
@@ -6614,27 +6572,33 @@ static int ssl_DecodePacketInternal(const byte* packet, int length, int isChain,
...
-        chainSz = (word32)length;
+        word32 chainSz = (word32)length;
 
-        chain = (struct iovec*)vChain;
+        chain = (struct iovec*)packet;
         length = 0;
-        for (i = 0; i < chainSz; i++)
+        for (i = 0; i < chainSz; i++) length += chain[i].iov_len;
+
+        tmpPacket = (byte*)XMALLOC(length, NULL, DYNAMIC_TYPE_TMP_BUFFER);
```
Comment (dgarske) [inline src/sniffer.c:6593]: I wonder if there is a better dynamic type like `DYNAMIC_TYPE_SNIFFER_PB_BUFFER` that could be used or maybe even a new `DYNAMIC_TYPE_SNIFFER_CHAIN_BUFFER`?

**Review by dgarske — CHANGES_REQUESTED** [review]
Otherwise PR looks awesome!

**Review by dgarske — APPROVED** [review]
Please squash this PR. Thanks @bandi13

**Comment by bandi13** [comment]
> Can you update the comment: "Please run directly from wolfSSL root dir".... It is no longer true.
> 
> ```
> Enter the port to scan [default: 11111, '0' for all]:
> Enter the server key [default: ../../certs/statickeys/ecc-secp256r1.pem,../../certs/statickeys/dh-ffdhe2048.pem,../../certs/statickeys/x25519.pem]:
> Enter alternate SNI [default: none]:
> Failed loading private key ../../certs/statickeys/ecc-secp256r1.pem: ret -1
> Please run directly from wolfSSL root dir
> Failed loading private key ../../certs/statickeys/dh-ffdhe2048.pem: ret -1
> Please run directly from wolfSSL root dir
> Failed loading private key ../../certs/statickeys/x25519.pem: ret -1
> Please run directly from wolfSSL root dir
> davidgarske@Davids-MacBook-Pro-2 wolfssl % pwd
> /Users/davidgarske/GitHub/wolfssl
> ```

Done, and squashed.

**Review by dgarske — APPROVED** [review]

**Pull request merged**